packages/artifact/src/artifact.ts

```diff
@@ -0,0 +1,108 @@
+import * as core from '@actions/core'
+import {CreateArtifactResponse} from './contracts'
+import {SearchResult, findFilesToUpload} from './search'
+import {
+  createArtifactInFileContainer,
+  uploadArtifactToFileContainer,
+  patchArtifactSize
+} from './upload-artifact-http-client'
+import {UploadInfo} from './upload-info'
+import {checkArtifactName} from './utils'
+
+/**
+ * Uploads an artifact
+ *
+ * @param name the name of the artifact, required
+ * @param path the directory, file, or glob pattern to denote what will be uploaded, required
+ * @returns single UploadInfo object
+ */
+export async function uploadArtifact(
+  name: string,
+  path: string
+): Promise<UploadInfo> {
+  // Check that the required inputs are valid
+  if (!name) {
+    throw new Error('Artifact name must be provided')
+  }
+
+  checkArtifactName(name)
+
+  if (!path) {
+    throw new Error('Upload path must be provided')
+  }
+
+  // Search for the items that will be uploaded
+  const filesToUpload: SearchResult[] = await findFilesToUpload(name, path)
+  let reportedSize = -1
+
+  if (filesToUpload === undefined) {
+    core.setFailed(
+      `Unable to succesfully search fo files to upload with the provided path: ${path}`
+    )
+  } else if (filesToUpload.length === 0) {
+    core.warning(
+      `No files were found for the provided path: ${path}. No artifacts will be uploaded.`
+    )
+  } else {
+    /**
+     * Step 1 of 3
+     * Create an entry for the artifact in the file container
+     */
+    const response: CreateArtifactResponse = await createArtifactInFileContainer(
+      name
+    )
+    if (!response.fileContainerResourceUrl) {
+      console.log(response)
+      throw new Error('Unable to get fileContainerResourceUrl')
+    }
+    core.debug(`We will be uploading to: ${response.fileContainerResourceUrl}`)
```

Comment:
nit: Can be more concise with just `Upload Resource URL: ${response.fileContainerResourceUrl}`

"We" feels odd for debug logging

Reply:
Got rid of `we` and changed up the logging